packages/optimistic-oracle/src/proposer.js

```diff
@@ -314,19 +368,34 @@ class OptimisticOracleProposer {
           transactionConfig: { ...this.gasEstimator.getCurrentFastPrice(), from: this.account },
         });
 
-        const logResult = {
-          tx: receipt.transactionHash,
-          requester: receipt.events.DisputePrice.returnValues.requester,
-          proposer: receipt.events.DisputePrice.returnValues.proposer,
-          disputer: receipt.events.DisputePrice.returnValues.disputer,
-          identifier: this.hexToUtf8(receipt.events.DisputePrice.returnValues.identifier),
-          ancillaryData: receipt.events.DisputePrice.returnValues.ancillaryData || "0x",
-          timestamp: receipt.events.DisputePrice.returnValues.timestamp,
-          proposedPrice: receipt.events.DisputePrice.returnValues.proposedPrice,
-        };
+        let logResult;
+        if (this.optimisticOracleClient.oracleType === OptimisticOracleType.SkinnyOptimisticOracle) {
+          logResult = {
+            tx: receipt.transactionHash,
+            requester: receipt.events.DisputePrice.returnValues.requester,
+            proposer: receipt.events.DisputePrice.returnValues.request.proposer,
```

Comment:
This may be more work than it's worth, but for most of these logs, we could avoid the if entirely if we did something like this:
```js
const logResult = {
  tx: receipt.transactionHash,
  ...receipt.events.DisputePrice.returnValues.request, // This may be undefined sometimes, but that's fine!
  ...receipt.events.DisputePrice.returnValues,
  // We only need to manually list off any fields that require a transformation
  ancillaryData: receipt.events.DisputePrice.returnValues.ancillaryData || "0x",
  identifier: this.hexToUtf8(receipt.events.DisputePrice.returnValues.identifier)
};
```

I think this strategy should work for most, if not all, of the logs.

I'm totally fine leaving things the way they are, but just wanted to point this out in case you felt it was cleaner.

Reply:
Oh great call, this is much cleaner.